pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -401,6 +401,36 @@ func TestApplyTaskResults_MinimalExpression(t *testing.T) {
 				}},
 			},
 		}},
+	}, {
```

Comment:
Can you add several more test cases to test invalid and out of index cases?

Reply:
I added the validation in the latest commit,

Reply:
LGTM. You can resolve this